payu/models/access.py

```diff
@@ -208,13 +233,53 @@ def archive(self):
                     if os.path.exists(f_src):
                         shutil.move(f_src, f_dst)
 
-                # Copy configs from work path to restart
-                for f_name in model.config_files:
-                    f_src = os.path.join(model.work_path, f_name)
-                    f_dst = os.path.join(model.restart_path, f_name)
+                # Copy "cice_in.nml" from work path to restart.
+                work_ice_nml_path = os.path.join(
+                                        model.work_path,
+                                        model.ice_nml_fname
+                )
+                restart_ice_nml_path = os.path.join(
+                                        model.restart_path,
+                                        model.ice_nml_fname
+                )
 
-                    if os.path.exists(f_src):
-                        shutil.copy2(f_src, f_dst)
+                if os.path.exists(work_ice_nml_path):
+                    shutil.copy2(work_ice_nml_path, restart_ice_nml_path)
+
+            if model.model_type in ('cice', 'matm'):
+                # Write the simulation end date to the restart date
+                # namelist.
+
+                # Calculate the end date using information from the work
+                # directory coupling namelist.
+                work_cpl_fpath = os.path.join(model.work_path, model.cpl_fname)
+                work_cpl_nml = f90nml.read(work_cpl_fpath)
+                work_cpl_grp = work_cpl_nml[model.cpl_group]
+
+                # Timing information on the completed run.
+                run_init_date_int = work_cpl_grp["init_date"]
```

Comment:
Is it clearer to call this `exp_init_date...` ? Because its start date for the whole experiment ?

Reply:
Yep that's much clearer! I've added this in.